src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -1108,37 +1119,40 @@ def run(self):
-                    # run autofocus for the selected roi
-                    self._future.running_subf = autofocus_in_roi(roi,
-                                                                 self._stage,
-                                                                 self._ccd,
-                                                                 self._focus,
-                                                                 self.focus_rng,
-                                                                 self._focus_points[idx],
-                                                                 self.conf_level)
-
-                try:
-                    focus_points_per_area = len(self._focus_points[idx])
-                    max_wait_t = estimate_autofocus_in_roi_time(focus_points_per_area, self._ccd) * 3 + 1
-                    focus_points = self._future.running_subf.result(max_wait_t)
-                except TimeoutError:
-                    self.streams[0].is_active.value = False
-                    logging.debug(f"Autofocus timed out for roi number {idx}, with bounding box: {roi} [m]")
-                    raise
-                except Exception:
-                    self.streams[0].is_active.value = False
-                    logging.debug(f"Autofocus failed for roi number {idx}, with bounding box: {roi} [m]")
-                    raise
+                focus_points = None
+                if self._use_autofocus:
+                    # cancel the sub future
+                    with self._future._task_lock:
+                        if self._future._task_state == CANCELLED:
+                            raise CancelledError()
+                        # is_active set to True will keep the acquisition going on
+                        self.streams[0].is_active.value = True
+                        logging.debug(f"Autofocus is running for roi number {idx}, with bounding box: {roi} [m]")
+                        # run autofocus for the selected roi
+                        self._future.running_subf = autofocus_in_roi(roi,
+                                                                    self._stage,
+                                                                    self._det,
+                                                                    self._focus,
+                                                                    self.focus_rng,
+                                                                    self._focus_points[idx],
+                                                                    self.conf_level)
+
+                    try:
+                        focus_points_per_area = len(self._focus_points[idx])
+                        max_wait_t = estimate_autofocus_in_roi_time(focus_points_per_area, self._det) * 3 + 1
+                        focus_points = self._future.running_subf.result(max_wait_t)
+                    except TimeoutError:
+                        self.streams[0].is_active.value = False
```

Comment:
```suggestion
                        self.streams[0].is_active.value = False
```
Can this go in `finally`?

Reply:
good suggestion